File: tui/wiki/registry.go
package wiki

import (
	"encoding/json"
	"os"
	"path/filepath"
)

// Wiki represents a single registered wiki from ~/.your-ai-memory/config.json.
type Wiki struct {
	Name         string   `json:"name"`
	Path         string   `json:"path"`
	Domain       string   `json:"domain"`
	Topics       []string `json:"topics"`
	Created      string   `json:"created"`
	LastAccessed string   `json:"lastAccessed"`
}

type registry struct {
	Wikis []Wiki `json:"wikis"`
}

// ConfigPath returns the path to the global config file.
func ConfigPath() string {
	home, _ := os.UserHomeDir()
	return filepath.Join(home, ".your-ai-memory", "config.json")
}

// LoadRegistry reads ~/.your-ai-memory/config.json and returns all registered wikis.
// Returns an empty slice (not an error) if the file does not yet exist.
func LoadRegistry() ([]Wiki, error) {
	data, err := os.ReadFile(ConfigPath())
	if os.IsNotExist(err) {
		return []Wiki{}, nil
	}
	if err != nil {
		return nil, err
	}
	var reg registry
	if err := json.Unmarshal(data, &reg); err != nil {
		return nil, err
	}
	return reg.Wikis, nil
}

// SaveRegistry writes the wiki list back to config.json.
func SaveRegistry(wikis []Wiki) error {
	reg := registry{Wikis: wikis}
	data, err := json.MarshalIndent(reg, "", "  ")
	if err != nil {
		return err
	}
	configDir := filepath.Dir(ConfigPath())
	if err := os.MkdirAll(configDir, 0755); err != nil {
		return err
	}
	return os.WriteFile(ConfigPath(), data, 0644)
}

// UpdateLastAccessed updates the lastAccessed field for a wiki by name.
func UpdateLastAccessed(wikis []Wiki, name string) []Wiki {
	today := todayString()
	for i := range wikis {
		if wikis[i].Name == name {
			wikis[i].LastAccessed = today
		}
	}
	return wikis
}

// GroupByTopic groups wikis by their first topic tag.
func GroupByTopic(wikis []Wiki) map[string][]Wiki {
	groups := make(map[string][]Wiki)
	for _, w := range wikis {
		tag := "untagged"
		if len(w.Topics) > 0 {
			tag = w.Topics[0]
		}
		groups[tag] = append(groups[tag], w)
	}
	return groups
}

func todayString() string {
	// Use time package via caller — keeping this package dependency-free.
	// Callers that need today's date should use time.Now().Format("2006-01-02").
	return ""
}

File: cli/lib/registry.js
import fs from 'fs-extra';
import path from 'path';
import os from 'os';

const CONFIG_DIR = path.join(os.homedir(), '.your-ai-memory');
const CONFIG_PATH = path.join(CONFIG_DIR, 'config.json');

export async function readRegistry() {
  try {
    await fs.ensureDir(CONFIG_DIR);
    if (!await fs.pathExists(CONFIG_PATH)) {
      return { wikis: [] };
    }
    return await fs.readJson(CONFIG_PATH);
  } catch {
    return { wikis: [] };
  }
}

export async function writeRegistry(data) {
  await fs.ensureDir(CONFIG_DIR);
  await fs.writeJson(CONFIG_PATH, data, { spaces: 2 });
}

export async function registerWiki(entry) {
  const registry = await readRegistry();
  const existing = registry.wikis.findIndex(w => w.path === entry.path);
  if (existing >= 0) {
    registry.wikis[existing] = { ...registry.wikis[existing], ...entry };
  } else {
    registry.wikis.push(entry);
  }
  await writeRegistry(registry);
}

export async function updateLastAccessed(name) {
  const registry = await readRegistry();
  const wiki = registry.wikis.find(
    w => w.name === name || w.name.toLowerCase() === name.toLowerCase()
  );
  if (wiki) {
    wiki.lastAccessed = new Date().toISOString().slice(0, 10);
    await writeRegistry(registry);
  }
  return wiki;
}

export async function findWiki(nameOrPath) {
  const registry = await readRegistry();
  return registry.wikis.find(
    w =>
      w.name === nameOrPath ||
      w.name.toLowerCase() === nameOrPath.toLowerCase() ||
      w.path === nameOrPath
  );
}
